utilities/postgres.go

```diff
@@ -0,0 +1,67 @@
+package utilities
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+)
+
+// PostgresError is a custom error struct for marshalling Postgres errors to JSON.
+type PostgresError struct {
+	Code           string `json:"code"`
+	HttpStatusCode int    `json:"-"`
+	Message        string `json:"message"`
+	Hint           string `json:"hint,omitempty"`
+	Detail         string `json:"detail,omitempty"`
+}
+
+// NewPostgresError returns a new PostgresError if the error was from a publicly
+// accessible Postgres error.
+func NewPostgresError(err error) *PostgresError {
+	var pgErr *pgconn.PgError
+	if errors.As(err, &pgErr) && isPubliclyAccessiblePostgresError(pgErr.Code) {
+		return &PostgresError{
+			Code:           pgErr.Code,
+			HttpStatusCode: getHttpStatusCodeFromPostgresErrorCode(pgErr.Code),
+			Message:        pgErr.Message,
+			Detail:         pgErr.Detail,
+			Hint:           pgErr.Hint,
+		}
+	}
+
+	return nil
+}
+
+// isPubliclyAccessiblePostgresError checks if the Postgres error should be
+// made accessible.
+func isPubliclyAccessiblePostgresError(code string) bool {
```

Comment:
Not sure about the naming of this: maybe `isUserTriggeredPostgresError` might be better?

Reply:
yeah that sounds good, although i think the current naming is fine too - some other suggestions i have are `isExternalPostgresError`, `isPgErrCode`?